lib/experimental/html/class-wp-html-tag-processor.php

```diff
@@ -1099,14 +1175,104 @@ private function apply_attributes_updates() {
+			 * $position->start and $position->end to $diff->start. We can't
+			 * change it and still expect the correct result, so let's accumulate
+			 * the deltas separately and apply them all at once after the loop.
+			 */
+			$head_delta = 0;
+			$tail_delta = 0;
+
+			foreach ( $this->attribute_updates as $diff ) {
+				$update_head = $position->start >= $diff->start;
+				$update_tail = $position->end >= $diff->start;
+
+				if ( ! $update_head && ! $update_tail ) {
+					break;
+				}
+
+				$delta = strlen( $diff->text ) - ( $diff->end - $diff->start );
+
+				if ( $update_head ) {
+					$head_delta += $delta;
+				}
+
+				if ( $update_tail ) {
+					$tail_delta += $delta;
+				}
+			}
+
+			$position->start += $head_delta;
+			$position->end   += $tail_delta;
+		}
+
 		$this->attribute_updates = array();
 	}
 
+	/**
+	 * Move the current pointer in the Tag Processor to a given bookmark's location.
+	 *
+	 * The maximum number of seek() calls is restricted to seek_limit() to protect
+	 * against infinite loops. If you need to process a very large document, adjust
+	 * the limit like `$processor->seek_limit( $new_limit );`.
+	 *
+	 * @param string $bookmark_name Name of bookmark to which to rewind.
+	 * @return bool
+	 * @throws Exception Throws on invalid bookmark name if WP_DEBUG set.
+	 */
+	public function seek( $bookmark_name ) {
+		if ( ! array_key_exists( $bookmark_name, $this->bookmarks ) ) {
+			if ( defined( 'WP_DEBUG' ) && WP_DEBUG ) {
+				throw new Exception( 'Invalid bookmark name' );
+			}
+			return false;
+		}
+
+		/**
+		 * A fail-safe that short-circuits in case the number of seek() calls
+		 * is suspiciously high. It could mean the user-code got into an infinite
+		 * loop somehow.
+		 */
+		++$this->seek_counter;
+		if($this->seek_counter > $this->max_seek_calls) {
+			if ( defined( 'WP_DEBUG' ) && WP_DEBUG ) {
+				throw new Exception( "Too many seek() calls ({$this->seek_counter}) – did you run into an infinite loop accidentally? If not, you may adjust this via the `seek_limit( \$new_limit )` method." );
+			}
+			return false;
+		}
+
+		$this->apply_html_updates_and_stay_on_current_tag();
+
+		// Point this tag processor before the sought tag opener and consume it
+		$this->parsed_bytes  = $this->bookmarks[ $bookmark_name ]->start;
+		$this->updated_bytes = $this->parsed_bytes;
+		$this->updated_html  = substr( $this->html, 0, $this->updated_bytes );
+		return $this->next_tag();
+	}
+
+	/**
+	 * Returns, and optionally updates, the maximum allowed number of seek() calls.
+	 *
+	 * The limit is a fail-safe to protect against accidental infinite loops.
+	 *
+	 * @see seek
+	 * @param int $new_limit Optional. The maximum allowed number of seek() calls.
+	 * @return number The allowed number of seek() calls.
+	 */
+	public function seek_limit( $new_limit=null ) {
+		if(is_int($new_limit)) {
+			$this->max_seek_calls = $new_limit;
+		}
+		return $this->max_seek_calls;
+	}
```

Comment:
given an optional value I would expect this function to reset the limit to the original default value.

```php
return $this->max_seek_calls = is_int( $new_limit ) ? $new_limit : self::MAX_SEEK_OPERATIONS;
```

Reply:
I'd rather remove the default argument and do `set_max_seek_calls()`  and `get_max_seek_calls()`

Reply:
is there a problem simply leaving it exposed as a public property? no need for `get_` and `set_` functions

Reply:
A public property sounds great